### DISCUSSION TITLE
Bring specifc item into view (VirtualScrollView)

### DISCUSSION THREAD

**Discussion by ghost** [marc2332/freya#956] [Q&A] [answered]
Hi,

I'm using the `VirtualScrollView` and have implemented the following behavior. I can select an item in a list. I can move the selection with arrow key up and arrow key down. What I want now is that the selected item is always visible when I move the selection. I have seen for custom scroll behavior I can use `ScrollController`. But I'm not sure how I can read the the `y` position of the selected item to check if is currently visible or not. Any ideas?

**Comment by marc2332** [comment]
> But I'm not sure how I can read the the y position of the selected item to check if is currently visible or not. 
 
You can calculate the `y` of an item by simply multiplying its `index` * `<item_size>` and then making it negative, example:

- item index is `58`
- item size is `35`
- item y is then: `-(58 x 35)` = `-2610`

if you scroll to the `-2160` you should be able to see the item on top, for this you will need to use a scroll controller

hope it helps!

**Reply by ghost** [reply]
Hi thank you the response.

The thing to get y of an item is clear, what is missing for me is to read the current y scroll offset. To calculate the current y value of an item if the content is scrolled.

**Reply by marc2332** [reply, marked as answer]
I just added a `x` and `y` getters to the scroll controller, that should let you easily get the current scroll position :D

**Reply by ghost** [reply]
Awesome thank you :-)

**Reply by ghost** [reply]
That was fast ;-)

By the way this is what I am working on.
![2024-10-04T15:34:22,973445793+02:00](https://github.com/user-attachments/assets/5598ae0d-40c6-4cd6-b1db-ee423f9182dd)

It's an application launcher for Linux. It is already usable.

**Reply by marc2332** [reply]
That looks cool!

**Reply by ghost** [reply]
Thank you :-)

**Comment by ghost** [comment]
@marc2332 one other question. How to get the actual `height` of the `VirtualScrollView` I need it to check if `item_y + item_height > scroll_view_height`.

**Reply by marc2332** [reply]
mmm, I think something like `-(<item-size> * <item-len>)` should be it

**Reply by ghost** [reply]
What I need is the visible height of the `VirtualScrollView` I need to check if I scroll if the item is partial of full outside of the visible area.

**Reply by ghost** [reply]
@marc2332 maybe something like `use_node` for a child node?

**Reply by marc2332** [reply]
Ah yes, you can wrap the virtual scroll view inside a rect and use use_node in the rect, the virtual scroll view will expand to the rect size, so it should be fine
